CommandLine.c

```diff
@@ -204,12 +205,15 @@ static CommandLineStatus parseArguments(int argc, char** argv, CommandLineSettin
             if (!username) {
                flags->userId = geteuid();
             } else if (!Action_setUserOnly(username, &(flags->userId))) {
-               for (const char* itr = username; *itr; ++itr)
-                  if (!isdigit((unsigned char)*itr)) {
+               char *endptr;
+               errno = 0;
+               unsigned long res = strtoul(username, &endptr, 10);
+               unsigned castRes = (unsigned) res;
+               if (*endptr != '\0' || res == ULONG_MAX || errno != 0 || castRes != res) {
```

Comment:
```suggestion
               if (*endptr != '\0' || res == ULONG_MAX || errno != 0 || res > UINT_MAX) {
```
or
```suggestion
               if (*endptr != '\0' || res > UINT_MAX || errno != 0) {
```

Reply:
@BenBE  I would say `res >= UINT_MAX` because, AFAIK, (uid_t)-1 is invalid.

Reply:
That's something the current version didn't check for …